### PR TITLE
[DO-552] Fix payment initiation flow

### PR DESCRIPTION
**Pull request by Hdd87**
updated Payment Initiation flow diagram as per comments

### CODE REVIEW

**Review by d0rich — CHANGES_REQUESTED** [review]
## 1. Too much nested alternatives

It is really hard too keep in mind all alternatives when I read diagram. Delete some alternatives. For example, for me it is obvious that that flow will stop if (pre)validation is not passed.

image

## 2. Delete flows names

It is not technical peace of docs, so flows names are redundant.

image

## 3. Remove immediate settlement

Immediate settlement is extremely rare situation in real world, so it is better to delete this information. Moreover, flow without immediate settlement does not contradict with flow with immediate settlement.

image

## 4. Status flow

"pending and Accepted or Rejected" - what does this mean?

**Review by d0rich — CHANGES_REQUESTED** [review]
1. I think it is better to delete alt frame for prevalidation.
2. The same about alt frame for validation. We just don't know what is going on if validation fails. I think providing only positive scenario will be best solution here.
3. Validate request - business action. Update state - technical detail. I would delete "update state" to keep same level of abstraction.
4. I would add 1 business step to make it clear what is going on between pending and accepted/rejected

image

File: content/3.workflow/4.payment-initiation.md [hunk 1]
```diff
@@ -6,50 +6,55 @@
...
+
+    %% after validation by Gateway
+
+    gpg ->> psp: payment instruction
+    gpg ->> pn: Payment Status: Sent to Gateway <br/> corda state updated
+
+    %% Pending Status by PSP
+    alt PSP sends pending status for Payment Instruction
+    psp ->> gpg: Payment Status: Pending
+    gpg ->> pn : Payment Status: status from PSP passed to corDapp <br/> corda state updated
```
Comment (lingvolab) [inline content/3.workflow/4.payment-initiation.md:51]:
```suggestion
    gpg ->> pn : Update Corda state <br/> Payment Status: Pending 
```

File: content/3.workflow/4.payment-initiation.md [hunk 2]
```diff
@@ -6,50 +6,55 @@
...
+    
+    %% Request has been processed by PSP 
+    else PSP processed Payment Instruction
```
Comment (lingvolab) [inline content/3.workflow/4.payment-initiation.md:54]:
```suggestion
    else PSP processes Payment Instruction
```

File: content/3.workflow/4.payment-initiation.md [hunk 3]
```diff
@@ -6,50 +6,55 @@
...
+    psp ->> gpg: Payment Status: Accepted or Rejected
```
Comment (lingvolab) [inline content/3.workflow/4.payment-initiation.md:55]:
```suggestion
    psp ->> gpg: Payment Status: Accepted/Rejected
```

File: content/3.workflow/4.payment-initiation.md [hunk 4]
```diff
@@ -6,50 +6,55 @@
...
+    gpg ->> pn : Payment Status: status from PSP passed to corDapp <br/> corda state updated
```
Comment (lingvolab) [inline content/3.workflow/4.payment-initiation.md:56]:
```suggestion
    gpg ->> pn : Update Corda state <br/> Payment Status: Accepted/Rejected 
```

File: content/3.workflow/4.payment-initiation.md [hunk 5]
```diff
@@ -6,50 +6,55 @@
...
-    - Node Identity – whether a payment request is received from the node that initially registered the Participant initiating the request;
-    - Participant ID – whether the Participant is registered with the Payments network and if Participant ID is registered to the Participant’s node;
+    The Payments Gateway checks if the Participant is authorized to submit this payment request, validating the payment instruction from the following aspects:
+    - Node Identity – whether a payment request is received from the node that initially registered the Participant initiating the request.
+    - Participant ID – whether the Participant is registered with the Payments network and if Participant ID is registered to the Participant’s node.
 
-5. Once the PSP replies to the payment request, a returned payment status (_Accepted_, _Pending_ or _Rejected_) is propagated back to the Participant’s node. The updated status is made available as a new Corda state in the payment statuses chain.
+5. If valid, the Payment Gateway sends the payment instruction to the external gateway of the PSP.
```
Comment (lingvolab) [inline content/3.workflow/4.payment-initiation.md:15]:
```suggestion
5. If valid, the Payment Gateway sends the payment instruction to the external gateway of the PSP, updates the status of the payment instruction to _Set to Gateway_ and communicates it to the Participant.

```

**Review by lingvolab — CHANGES_REQUESTED** [review]

File: content/3.workflow/4.payment-initiation.md [hunk 6]
```diff
@@ -6,50 +6,55 @@
...
-
 ```mermaid
 ---
 title: Initiate Payment
 ---
 sequenceDiagram
-   
-    participant pcd as Payment CorDapp<br/> in Participant Node
+    %% Participants
+    participant pn as Payments Node <br/> with Custom CorDapp and Payments SDK
```
[the remark this answers is not in the document]
Comment (Hdd87) [reply content/3.workflow/4.payment-initiation.md:30]: point 2 and information in  presentation : where the payment node has CorDapp and an SDK

**Review by d0rich — APPROVED** [review]

File: content/3.workflow/4.payment-initiation.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (lingvolab) [reply content/3.workflow/4.payment-initiation.md:31]: Let's change it to "Participant Node" (without a second line)

[the remark this answers is not in the document]
Comment (Hdd87) [reply content/3.workflow/4.payment-initiation.md]: @lingvolab   fixed

**Review by lingvolab — APPROVED** [review]

**Pull request merged**